src/Microsoft.NetCore.Analyzers/Core/MicrosoftNetCoreAnalyzersResources.resx

```diff
@@ -1179,4 +1179,13 @@
   <data name="MaybeDisableHttpClientCRLCheckMessage" xml:space="preserve">
     <value>HttpClient may be created without enabling CheckCertificateRevocationList</value>
   </data>
+  <data name="MutableStructsShouldNotBeUsedForReadonlyFieldsTitle" xml:space="preserve">
+    <value>Certain readonly structs should not be used as readonly fields</value>
+  </data>
+  <data name="MutableStructsShouldNotBeUsedForReadonlyFieldsDescription" xml:space="preserve">
+    <value>Certain value types (e.g. SpinLock, GCHandle) are explicitly designed to be mutable. Declaring a field of such type as readonly might cause unintentional shadow copying by the compiler, leading to performance degradation or bugs.</value>
```

Comment:
Given that `SpinLock` and `GCHandle` are just couple of potentially many such types for which this rule can be applicable, I think we should make this configurable for end users to allow specifying the set of disallowed types such that mutable readonly fields of all of those types whould be flagged. If we agree, then I would suggest the following:

1. Use the [existing option](https://github.com/dotnet/roslyn-analyzers/blob/master/docs/Analyzer%20Configuration.md#disallowed-symbol-names) `disallowed_symbol_names`
2. See how this option is used to make CA1031 configurable with this commit: https://github.com/dotnet/roslyn-analyzers/commit/40cd30186631a3e5d33492b37f5c69b0aa59876c#diff-078582649f9598152821c2fd110c7219. I would suggest a similar implementation and unit test for it.

If you feel this would overload your PR, I can do this as a followup PR as well. In this case, please file a separate tracking issue to make this rule configurable.

Reply:
I agree that configuring this could be useful but I'd opt into enabling this configuration in a separate PR. I'll file it once this one is merged, if that's ok.

Reply:
Sounds reasonable!